src/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs

```diff
@@ -629,8 +629,31 @@ private bool StartWithCreateProcess(ProcessStartInfo startInfo)
             return ret;
         }
 
-        private static Encoding GetEncoding(int codePage)
+        private static Encoding GetConsoleEncodingOrThrow(ConsoleStreamDirection encodingType)
         {
+            const int INVALID_HANDLE = 6;
+            int codePage;
+            try
+            {
+                codePage = (encodingType == ConsoleStreamDirection.Input) ?
+                    (int)Interop.Kernel32.GetConsoleCP() :
+                    (int)Interop.Kernel32.GetConsoleOutputCP();
+
+                int error = Marshal.GetLastWin32Error();
+                if (error == INVALID_HANDLE)
+                {
+                    throw new PlatformNotSupportedException(SR.ConsoleEncodingNotSupported);
+                }
+                else if (error != 0)
+                {
+                    throw new InvalidOperationException(SR.ConsoleEncodingNotDetected);
```

Comment:
Prevoiusly we weren't checking the error code. Is this potentially a breaking change in some situation - where somehow it was previously working even though it was returning an error?